github/git_refs.go

```diff
@@ -56,15 +59,69 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 		return nil, nil, err
 	}
 
-	r := new(Reference)
-	resp, err := s.client.Do(ctx, req, r)
+	buf := new(bytes.Buffer)
```

Comment:
To clarify, I meant to ask if you can keep the code as before, have it `r := new(Reference); resp, err := s.client.Do(ctx, req, r)`? Or does that obscure the `*json.UnmarshalTypeError`?